### ISSUE TITLE
Resolve object API HTTP endpoint broken when passed query

### ISSUE THREAD

**Issue by SleeplessOne1917** [LemmyNet/lemmy#2836]
When working on an issue for the UI repo, I got the following error when trying to make an HTTP request to the `resolve_object` endpoint.
`404: FetchError: invalid json response body at http://lemmy:8536/api/v3/resolve_object?q=foo reason: Unexpected token 'Q', "Query dese"... is not valid JSON`.
It worked correctly back when I was working on [this UI issue](https://github.com/LemmyNet/lemmy-ui/issues/968). Were there any recent changes to how `resolve_object` is handled that would cause this to break?

**Comment by Nutomic** [comment]
Can you post the full message that is returned? And is there any error in the Lemmy logs?

**Comment by Nutomic** [comment]
Yes that is correct. This endpoint could be abused by fetching lots of remote objects which are stored forever, so its better to require auth.

**Comment by SleeplessOne1917** [comment]
Gotcha. This must be an issue with the UI project not including auth in its requests then. I'll create an issue over there.

**Comment by SleeplessOne1917** [comment]
I just realized there might actually be some work still required for the server side. Endpoints that require auth but aren't provided it should probably return a response with a 401 Unauthorized error code.

**Comment by Nutomic** [comment]
Right we arent currently doing that because lemmy-ui uses websocket so there are no HTTP status codes. https://github.com/LemmyNet/lemmy/pull/2819 is going in that direction, so I might also add the error 401.

**Comment by SleeplessOne1917** [comment]
Thanks for being so responsive. If the 401 status code is going to be part of your linked PR, I'll close this issue.
